solver/src/interactions/weth.rs

```diff
@@ -16,6 +19,25 @@ impl Interaction for UnwrapWethInteraction {
     }
 }
 
+impl UnwrapInteraction for UnwrapWethInteraction {
+    fn token_address(&self) -> H160 {
+        self.weth.address()
+    }
+
+    fn amount(&self) -> U256 {
+        self.amount
+    }
+
+    fn increase_amount(&mut self, amount: U256) {
```

Comment:
nit: Maybe a "can_merge(&self, other: &UnwrapInteraction)" logic would make sense which could check that the token addresses are the same and could also skip merging if it was to overflow?

Reply:
That's a really nice suggestion. The reason I had it like this was to use a token hash-map for unwrapping, but searching through a vector linearly would be fine as well given the expected size, and clean up the abstraction as well.